core-api/src/main/java/com/optimizely/ab/bucketing/Bucketer.java

```diff
@@ -102,6 +109,19 @@ private Variation bucketToVariation(@Nonnull Experiment experiment,
                                         @Nonnull String userId) {
         // "salt" the bucket id using the experiment id
         String combinedBucketId = userId + experiment.getId();
+
+        // If a persistent bucketer instance is present then check it for a persisted variation
+        if (persistentBucketer != null) {
+            Variation variation = persistentBucketer.restoreActivation(projectConfig, userId, experiment);
+            if (variation != null) {
+                logger.info("Returning previously activated variation \"{}\" from persistent bucketer.", variation.getKey());
+                // A variation is stored for this combined bucket id
+                return variation;
```

Comment:
What if this variation is not valid any more and was removed from customer's code? I think in that case we should check against the datafile to determine if variation is still valid.